asn1crypto/core.py

```diff
@@ -481,6 +490,12 @@ def __unicode__(self):
 
         return self.__repr__()
 
+    def __reduce__(self):
```

Comment:
Upon review of the general `pickle` API, I decided to stick with `__reduce__` as opposed to `__getstate__` and `__setstate__`, the reason being that implementing `__setstate__` would involve reapplying the state of an `asn1crypto` object _after_ it has been created. This is more complicated than it sounds, because `pickle` doesn't call `__init__` on object creation in typical circumstances. The `__setstate__` implementation would have to take that into account. IMHO that's more trouble than it's worth, and would potentially require a lot of extra tests to make sure all `asn1crypto` types are properly pickled and unpickled.

On the other hand, implementing `__reduce__`, while it involves a lower-level API, is much easier in this case: `__reduce__` allows the implementer to not only control object state, but also object creation, which allows us to get away with simply calling the relevant class's `.load(...)` method on the pickled data. Bonus: `.dump()` and `.load()` are already extensively tested.

Summary: I believe the `__reduce__` approach is much simpler and less error-prone in this particular situation. :)

Thanks again for the code review!